pymedphys/labs/quickcheck/qcheck.py

```diff
@@ -0,0 +1,180 @@
+            try:
+                self.sock.sendto(self.MSG, (self.ip, self.port))
+                self.raw_data, _ = self.sock.recvfrom(4096)
+                data = self.raw_data.decode(encoding="utf-8")
+                self.data = data.strip("\r\n")
+                break
+            except socket.timeout:
+                if n_retry == max_retries:
+                    print(
+                        """
+                          Connection Error  - Reached max retries
+                          Quickcheck device unreachable, please check your settings"""
+                    )
+                    self.data = ""
+                    break
+
+                print("Connection Timeout")
+                n_retry += 1
+                print("Retrying connection {}/{}".format(n_retry, max_retries))
+
+    def parse_measurements(self):
+        data_split = self.data.split(";")
+        m = {}  # Dictionary with measurements
+        if data_split[0] == "MEASGET":
+            #  MD section:__________________________________________________________
+            MD = re.findall(r"MD=\[(.*?)\]", self.data)[0]
+            m["MD_ID"] = np.int(re.findall(r"ID=(.*?);", MD)[0])
+            meas_date = re.findall(r"Date=(.*?);", MD)[0]
+            m["MD_Date"] = datetime.datetime.strptime(meas_date, "%Y-%m-%d").date()
+            meas_time = re.findall(r"Time=(.*?)$", MD)[0]
+            m["MD_Time"] = datetime.datetime.strptime(meas_time, "%H:%M:%S").time()
+            m["MD_DateTime"] = datetime.datetime.combine(m["MD_Date"], m["MD_Time"])
+            #  MV section:__________________________________________________________
+            str_val = re.findall(r"MV=\[(.*?)\]", self.data)[0]
+            m["MV_CAX"] = np.float(re.findall("CAX=(.*?);", str_val)[0])
+            m["MV_G10"] = np.float(re.findall("G10=(.*?);", str_val)[0])
+            m["MV_L10"] = np.float(re.findall("L10=(.*?);", str_val)[0])
+            m["MV_T10"] = np.float(re.findall("T10=(.*?);", str_val)[0])
+            m["MV_R10"] = np.float(re.findall("R10=(.*?);", str_val)[0])
+            m["MV_G20"] = np.float(re.findall("G20=(.*?);", str_val)[0])
+            m["MV_L20"] = np.float(re.findall("L20=(.*?);", str_val)[0])
+            m["MV_T20"] = np.float(re.findall("T20=(.*?);", str_val)[0])
+            m["MV_R20"] = np.float(re.findall("R20=(.*?);", str_val)[0])
+            m["MV_E1"] = np.float(re.findall("E1=(.*?);", str_val)[0])
+            m["MV_E2"] = np.float(re.findall("E2=(.*?);", str_val)[0])
+            m["MV_E3"] = np.float(re.findall("E3=(.*?);", str_val)[0])
+            m["MV_E4"] = np.float(re.findall("E4=(.*?);", str_val)[0])
+            m["MV_Temp"] = np.float(re.findall("Temp=(.*?);", str_val)[0])
+            m["MV_Press"] = np.float(re.findall("Press=(.*?);", str_val)[0])
+            m["MV_CAXRate"] = np.float(re.findall("CAXRate=(.*?);", str_val)[0])
+            m["MV_ExpTime"] = np.float(re.findall("ExpTime=(.*?)$", str_val)[0])
+
```

Comment:
As an interesting note, I find the above pattern is often more neatly expressed in the following way:

```python
regex_map = {
	"MV_CAX": "CAX=(.*?);",
    "MV_G10": "G10=(.*?);",
    ...
}

for key, pattern in regex_map.items():
    m[key] = np.float(re.findall(pattern, str_val)[0])
```

Reply:
That looks better indeed. Let me try to change it, and the logo stuff too. Thanks!